### PR TITLE
Addition of JDK8 module (fixes issue #982)

### PR DESCRIPTION
**Pull request by lpandzic**
No tests yet - it is currently just a proof of concept. Comments?

### CODE REVIEW

**Comment by lpandzic** [comment]
We'll need a job with jdk 8 to test this one.

File: pom.xml [hunk 1]
```diff
@@ -47,12 +51,12 @@
     <dependency>
       <groupId>com.fasterxml.jackson.core</groupId>
       <artifactId>jackson-annotations</artifactId>
-      <version>2.7.0-SNAPSHOT</version>
+      <version>${jackson.version}</version>
```
Comment (cowtowncoder) [inline pom.xml]: This is not correct, as we intentionally want to keep annotations version at .0 (ideally, I'd only release minor versions like "2.6", but due to dev preferences, patch versions are released despite no changes going in).

**Comment by cowtowncoder** [comment]
I'll have to think about this a bit. Thank you for contributing it!

**Comment by lpandzic** [comment]
No problem, I'm glad to help.
This PR is mostly inspired by Spring way on those matters - for example - https://github.com/spring-projects/spring-framework/blob/master/spring-core/src/test/java/org/springframework/tests/JavaVersion.java.

File: pom.xml (hunk 1 above)
Comment (lpandzic) [inline pom.xml]: Fixed.

**Comment by lpandzic** [comment]
While testing this module I've detected a bug in parameter names module - here's a pull request with a fix - https://github.com/FasterXML/jackson-module-parameter-names/pull/27.

**Comment by lpandzic** [comment]
Master branch is unstable?
JDK 6 build failed due to missing nio.Path and JDK 7 failed with out of memory exception.

**Comment by lpandzic** [comment]
Any news regarding this PR?

**Comment by cowtowncoder** [comment]
Sorry, no. I will return to this after 2.7.0 release -- right now I want to resolve upgrade to Java 7 first.
But after that is done, can return to this question.

**Comment by lpandzic** [comment]
Any news? :)

**Comment by cowtowncoder** [comment]
After thinking it through, I don't think I want automatic inclusion of modules with the default.
Part of this is due to practical concerns (as of now, parameter names inclusion can have side effects, related to creator introspection problems that need to be eventually solved); and part of it is that from modelling perspective I want to keep things cleanly separate.

**Pull request closed without merging**